test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_trcomp_performance.py

```diff
@@ -0,0 +1,175 @@
+import pytest, re
+import numpy as np
+
+import boto3, sagemaker
+from sagemaker.tensorflow import TensorFlow
+from sagemaker.training_compiler.config import TrainingCompilerConfig
+
+from src.benchmark_metrics import (
+    TRCOMP_THRESHOLD,
+    get_threshold_for_image,
+)
+from test.test_utils import (
+    BENCHMARK_RESULTS_S3_BUCKET,
+    LOGGER,
+    get_framework_and_version_from_tag,
+    get_cuda_version_from_tag,
+)
+
+
+
+@pytest.fixture
+def num_gpus(instance_type):
+    if instance_type in [
+            'ml.p3.2xlarge',
+            'ml.g4dn.xlarge',
+            'ml.g4dn.2xlarge',
+            'ml.g4dn.4xlarge',
+            'ml.g4dn.8xlarge',
+            'ml.g4dn.16xlarge',
+            'ml.g5.xlarge',
+            'ml.g5.2xlarge',
+            'ml.g5.4xlarge',
+            'ml.g5.8xlarge',
+            'ml.g5.16xlarge',
+            ]:
+        return 1
+    elif instance_type in [
+            'ml.p3.16xlarge',
+            'ml.p3dn.24xlarge',
+            'ml.g5.48xlarge',
+            ]:
+        return 8
+    else:
+        raise ValueError(f'Unforeseen instance {instance_type}')
+
+
+@pytest.fixture
+def total_n_gpus(num_gpus, instance_count):
+    return num_gpus*instance_count
+
+
+@pytest.fixture
+def distribution_strategy(instance_type,  num_gpus, instance_count, request):
+    if instance_count==1:
+        if num_gpus==1:
+            request.applymarker(pytest.mark.one_device)
+            return 'one_device'
+        else:
+            request.applymarker(pytest.mark.mirrored)
+            return 'mirrored'
+    else:
+        request.applymarker(pytest.mark.multi_worker_mirrored)
+        return 'multi_worker_mirrored'
+
+
+@pytest.fixture
+def caching(instance_type,  num_gpus, instance_count, distribution_strategy):
+    if distribution_strategy in [
+                                'one_device',
+                                ]:
+        return False
+    else:
+        return True
+
+
+@pytest.fixture
+def sagemaker_session(region):
+    return sagemaker.Session(boto_session=boto3.Session(region_name=region))
+
+
+@pytest.fixture
+def framework_version(tensorflow_training):
+    _, version = get_framework_and_version_from_tag(tensorflow_training)
+    return version
+
+
+@pytest.fixture(autouse=True)
+def smtrcomp_only(framework_version, tensorflow_training, request):
+    short_version = float(".".join(framework_version.split('.')[:2]))
+    if short_version<2.9:
```

Comment:
Doing version comparisons as floats is fragile, because float(2.10) < float(2.9) even though Version(2.10) > Version(2.9). You can use SpecifierSet and Version as used elsewhere in the PR

https://github.com/aws/deep-learning-containers/blob/6796726c40c578bac999163e2837b4081ffc5a63/test/dlc_tests/sanity/test_pre_release.py#L604

Reply:
Fixed